samples/ClientSample/Tcp/TcpHubConnectionBuilderExtensions.cs

```diff
@@ -1,13 +1,18 @@
-﻿using System.Net;
+﻿using System;
+using System.Net;
 using ClientSample;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public static class TcpHubConnectionBuilderExtensions
     {
         public static IHubConnectionBuilder WithEndPoint(this IHubConnectionBuilder builder, IPEndPoint endPoint)
         {
-            builder.ConfigureConnectionFactory(() => new TcpConnection(endPoint));
+            Func<IConnection> connectionFactory = () => new TcpConnection(endPoint);
+            builder.Services.Replace(new ServiceDescriptor(typeof(Func<IConnection>), connectionFactory));
```

Comment:
We never use replace, we most use Add, last one wins.

Reply:
So `Add` will add or replace a service, and `Replace` also will add or replace a service? Huh

Ok